tensorflow/lite/kernels/strided_slice.cc

```diff
@@ -76,39 +76,39 @@ inline int32_t ClampedIndex(int32_t index, int dim, bool pos_stride) {
                           std::min(std::max(index, -dim), dim - 1), dim));
 }
 
-// TODO(b/77971377) this logic should be removed, as it's a duplication of
-// StartForAxis() & StopForAxis() in kernels/internal/reference/reference_ops.h
-inline int32_t GetBeginValueAtIndex(StridedSliceContext* op_context, int idx) {
-  const int dim = op_context->input->dims->data[idx];
-  const bool pos_stride = GetTensorData<int32_t>(op_context->strides)[idx] > 0;
-  return op_context->params->begin_mask & (1 << idx)
-             ? pos_stride ? 0 : dim - 1
-             : ClampedIndex(GetTensorData<int32_t>(op_context->begin)[idx], dim,
-                            pos_stride);
-}
-
-inline int32_t GetEndValueAtIndex(StridedSliceContext* op_context, int idx) {
-  const int dim = op_context->input->dims->data[idx];
-  const bool pos_stride = GetTensorData<int32_t>(op_context->strides)[idx] > 0;
-  return op_context->params->end_mask & (1 << idx)
-             ? pos_stride ? dim : -1
-             : ClampedIndex(GetTensorData<int32_t>(op_context->end)[idx], dim,
-                            pos_stride);
-}
-
 // Processes the indexing tensors (begin, end and strides) to resize the
 // output tensor. This function is callable from both Prepare() and Eval() as
 // long as the caller ensures the indexing tensors are present.
 TfLiteStatus ResizeOutputTensor(TfLiteContext* context,
                                 StridedSliceContext* op_context) {
   std::vector<int> output_shape_vector;
+  std::vector<int32_t> starts;
+  std::vector<int32_t> stops;
+  std::vector<int32_t> strides;
+
+  starts.reserve(GetTensorShape(op_context->begin).FlatSize());
+  stops.reserve(GetTensorShape(op_context->end).FlatSize());
+  strides.reserve(GetTensorShape(op_context->strides).FlatSize());
+
+  for (int idxs = 0; idxs < op_context->dims; ++idxs) {
```

Comment:
This logic is now duplicated with the start/stop/stride code in Eval, so it's questionable whether it's a net improvement. Can we try to consolidate?

Reply:
@jdduke , i know what you mean, actually we are creating two StridedSliceParams if there is a dynamic tensor detected while in Eval function, but the problem is that we have created these two StridedSliceParams with different parameters and hence producing different results, so consolidation of the two looks difficult. 
Do you have any suggestions for me which i can use to club them ?

I have two options : -
1. make the code common but still StridedSliceParams will be created twice, as the input params are different
2.  Create the StridedSliceParams only once and update the parameters of it instead of duplicating start,stop,strides.

Let me know your opinion on this.
Regards
Amit